File: app/src/main/java/dev/ragnarok/fenrir/mvp/view/IGiftsView.java
package dev.ragnarok.fenrir.mvp.view;

import java.util.List;

import dev.ragnarok.fenrir.model.Gift;
import dev.ragnarok.fenrir.mvp.core.IMvpView;
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView;


public interface IGiftsView extends IAccountDependencyView, IMvpView, IErrorView {
    void displayData(List<Gift> gifts);

    void notifyDataSetChanged();

    void notifyDataAdded(int position, int count);

    void showRefreshing(boolean refreshing);

    void onOpenWall(int accountId, int ownerId);
}

File: app_fenrir/src/main/kotlin/dev/ragnarok/fenrir/mvp/view/IGiftsView.kt
package dev.ragnarok.fenrir.mvp.view

import dev.ragnarok.fenrir.model.Gift
import dev.ragnarok.fenrir.mvp.core.IMvpView
import dev.ragnarok.fenrir.mvp.view.base.IAccountDependencyView

interface IGiftsView : IAccountDependencyView, IMvpView, IErrorView {
    fun displayData(gifts: List<Gift>)
    fun notifyDataSetChanged()
    fun notifyDataAdded(position: Int, count: Int)
    fun showRefreshing(refreshing: Boolean)
    fun onOpenWall(accountId: Int, ownerId: Int)
}